pkg/plugin/stream.go

```diff
@@ -39,29 +29,66 @@ func (a *App) SubscribeStream(ctx context.Context, req *backend.SubscribeStreamR
 }
 
 func (a *App) runOpenAIChatCompletionsStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
-	// Deserialize request data.
-	incomingBody := chatCompletionsRequest{Stream: true}
-	err := json.Unmarshal(req.Data, &incomingBody)
+
+	settings := loadSettings(*req.PluginContext.AppInstanceSettings)
+	requestBody := map[string]interface{}{}
+	var err error
+	err = json.Unmarshal(req.Data, &requestBody)
 	if err != nil {
-		return err
+		return fmt.Errorf("Unable to unmarshal request body: %w", err)
 	}
 
-	// Load app settings.
-	settings := loadSettings(*req.PluginContext.AppInstanceSettings)
+	// set stream to true
+	requestBody["stream"] = true
+
+	u, err := url.Parse(settings.OpenAI.URL)
+	if err != nil {
+		return fmt.Errorf("Unable to parse OpenAI URL: %w", err)
+	}
+
+	var outgoingBody []byte
+
+	if settings.OpenAI.UseAzure {
+		// Map model to deployment
 
-	// Create and send OpenAI request.
-	outgoingBody, err := json.Marshal(incomingBody)
+		var deployment string = ""
+		for _, v := range settings.OpenAI.AzureMapping {
+			if val, ok := requestBody["model"].(string); ok && val == v[0] {
+				deployment = v[1]
+				break
+			}
+		}
+
+		if deployment == "" {
+			return fmt.Errorf("No deployment found for model: %s", requestBody["model"])
+		}
+
+		u.Path = fmt.Sprintf("/openai/deployments/%s/chat/completion", deployment)
+		u.RawQuery = "api-version=2023-03-15-preview"
+
+		// Remove extra fields
+		delete(requestBody, "model")
+
+	} else {
+		u.Path = strings.TrimPrefix(req.Path, "openai")
```

Comment:
```suggestion
		u.Path = "/v1/chat/completions"
```

The request path might include a UUID on it in the suffix and we can just hardcode the chat completions endpoint.